frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-query-result/experiment-query-result.component.ts

```diff
@@ -46,14 +46,13 @@ export class ExperimentQueryResultComponent implements OnInit, OnDestroy {
 
   ngOnInit() {
     const queryIds = [];
-
+    // sort the factors:
+    this.experiment.factors = this.experiment.factors.slice().sort((a, b) => (a.order > b.order ? 1 : b.order > a.order ? -1 : 0));
```

Comment:
can this sort method please be pulled out into a class helper method? It's implemented twice in the same function. Something like

```typescript
sortFactorsByOrderAscending(factorA: ExperimentFactor, factorB: ExperimentFactor): number {
  factorA.order > factorB.order ? 1 : factorB.order > factorA.order ? -1 : 0)
}

so
```

Reply:
if `this.experiment.factors` is not an array, this would break; will `this.experiment.factors` still exist as an array if this component is loaded as a simple experiment?

Reply:
> can this sort method please be pulled out into a class helper method? It's implemented more than once. Something like
> 
> ```ts
> sortFactorsByOrderAscending(factorA: ExperimentFactor, factorB: ExperimentFactor): number {
>   factorA.order > factorB.order ? 1 : factorB.order > factorA.order ? -1 : 0)
> }
> ```

Sure, I will do that.

Reply:
> if `this.experiment.factors` is not an array, this would break; will `this.experiment.factors` still exist as an array if this component is loaded as a simple experiment?

It should not break, I will move it inside the check when the experiment is factorial.